packages/faucet/src/faucet.spec.ts

```diff
@@ -15,175 +13,165 @@ function pendingWithoutWasmd(): void {
   "economy stock theory fatal elder harbor betray wasp final emotion task crumble siren bottom lizard educate guess current outdoor pair theory focus wife stone";
 
-async function makeProfile(
-  distributors = 0,
-): Promise<{ readonly profile: UserProfile; readonly holder: Identity; readonly distributors: Identity[] }> {
-  const [profile, identities] = await createUserProfile(faucetMnemonic, defaultChainId, distributors);
-  return {
-    profile: profile,
-    holder: identities[0],
-    distributors: identities.slice(1),
-  };
-}
-
 describe("Faucet", () => {
   describe("constructor", () => {
     it("can be constructed", async () => {
       pendingWithoutWasmd();
-      const connection = await CosmosConnection.establish(httpUrl, defaultAddressPrefix, defaultConfig);
-      const { profile } = await makeProfile();
-      const faucet = new Faucet(defaultConfig, connection, codec, profile);
+      const faucet = await Faucet.make(httpUrl, defaultAddressPrefix, defaultTokenConfig, faucetMnemonic, 3);
       expect(faucet).toBeTruthy();
-      connection.disconnect();
+    });
+  });
+
+  describe("availableTokens", () => {
+    it("is empty when no tokens are configures", async () => {
+      const faucet = await Faucet.make(httpUrl, defaultAddressPrefix, { bankTokens: [] }, faucetMnemonic, 3);
+      const tickers = await faucet.availableTokens();
+      expect(tickers).toEqual([]);
+    });
+
+    it("is empty when no tokens are configures", async () => {
+      const faucet = await Faucet.make(httpUrl, defaultAddressPrefix, defaultTokenConfig, faucetMnemonic, 3);
+      const tickers = await faucet.availableTokens();
+      expect(tickers).toEqual(["COSM", "STAKE"]);
     });
   });
 
   describe("send", () => {
     it("can send bank token", async () => {
       pendingWithoutWasmd();
-      const connection = await CosmosConnection.establish(httpUrl, defaultAddressPrefix, defaultConfig);
-      const { profile, holder } = await makeProfile();
-      const faucet = new Faucet(defaultConfig, connection, codec, profile);
+      const faucet = await Faucet.make(httpUrl, defaultAddressPrefix, defaultTokenConfig, faucetMnemonic, 3);
       const recipient = makeRandomAddress();
       await faucet.send({
         amount: {
-          quantity: "23456",
-          fractionalDigits: 6,
-          tokenTicker: "COSM" as TokenTicker,
+          amount: "23456",
+          denom: "ucosm",
         },
-        sender: holder,
+        sender: faucet.holderAddress,
         recipient: recipient,
       });
-      const account = await connection.getAccount({ address: recipient });
+
+      const readOnlyClient = new CosmosClient(httpUrl);
+      const account = await readOnlyClient.getAccount(recipient);
       assert(account);
       expect(account.balance).toEqual([
         {
-          quantity: "23456",
-          fractionalDigits: 6,
-          tokenTicker: "COSM" as TokenTicker,
+          amount: "23456",
+          denom: "ucosm",
```

Comment:
okay, so we use the raw denom from the blockchain.

I think this is fine actually. Let's just leave it like this, updates the amounts in config (10COSM -> 10_000_000 ucosm) and wait until there is some magic metadata on-chain to handle this translation.

Reply:
This is a pretty low level function that takes a plain `Coin`. The higher level interface reads 10COSM from env variables and convers this internally